source/brailleDisplayDrivers/hid.py

```diff
@@ -0,0 +1,297 @@
+# brailleDisplayDrivers/HID.py
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2021 NV Access Limited
+
+from dataclasses import dataclass
+from typing import List, Optional
+import enum
+import braille
+import inputCore
+from logHandler import log
+import brailleInput
+import bdDetect
+import hidpi
+import hwIo.hid
+from hwIo import intToByte
+
+from bdDetect import HID_USAGE_PAGE_BRAILLE
+
+
+class BraillePageUsageID(enum.IntEnum):
+	UNDEFINED = 0
+	BRAILLE_DISPLAY = 0x1
+	BRAILLE_ROW = 0x2
+	EIGHT_DOT_BRAILLE_CELL = 0x3
+	SIX_DOT_BRAILLE_CELL = 0x4
+	NUMBER_OF_BRAILLE_CELLS = 0x5
+	SCREEN_READER_CONTROL = 0x6
+	SCREEN_READER_IDENTIFIER = 0x7
+	ROUTER_SET_1 = 0xFA
+	ROUTER_SET_2 = 0xFB
+	ROUTER_SET_3 = 0xFC
+	ROUTER_KEY = 0x100
+	ROW_ROUTER_KEY = 0x101
+	BRAILLE_BUTTONS = 0x200
+	BRAILLE_KEYBOARD_DOT_1 = 0x201
+	BRAILLE_KEYBOARD_DOT_2 = 0x202
+	BRAILLE_KEYBOARD_DOT_3 = 0x203
+	BRAILLE_KEYBOARD_DOT_4 = 0x204
+	BRAILLE_KEYBOARD_DOT_5 = 0x205
+	BRAILLE_KEYBOARD_DOT_6 = 0x206
+	BRAILLE_KEYBOARD_DOT_7 = 0x207
+	BRAILLE_KEYBOARD_DOT_8 = 0x208
+	BRAILLE_KEYBOARD_SPACE = 0x209
+	BRAILLE_KEYBOARD_LEFT_SPACE = 0x20A
+	BRAILLE_KEYBOARD_RIGHT_SPACE = 0x20B
+	BRAILLE_FACE_CONTROLS = 0x20C
+	BRAILLE_LEFT_CONTROLS = 0x20D
+	BRAILLE_RIGHT_CONTROLS = 0x20E
+	BRAILLE_TOP_CONTROLS = 0x20F
+	BRAILLE_JOYSTICK_CENTER = 0x210
+	BRAILLE_JOYSTICK_UP = 0x211
+	BRAILLE_JOYSTICK_DOWN = 0x212
+	BRAILLE_JOYSTICK_LEFT = 0x213
+	BRAILLE_JOYSTICK_RIGHT = 0x214
+	BRAILLE_DPAD_CENTER = 0x215
+	BRAILLE_DPAD_UP = 0x216
+	BRAILLE_DPAD_DOWN = 0x217
+	BRAILLE_DPAD_LEFT = 0x218
+	BRAILLE_DPAD_RIGHT = 0x219
+	BRAILLE_PAN_LEFT = 0x21A
+	BRAILLE_PAN_RIGHT = 0x21B
+	BRAILLE_ROCKER_UP = 0x21C
+	BRAILLE_ROCKER_DOWN = 0x21D
+	BRAILLE_ROCKER_PRESS = 0x21E
+
+
+@dataclass
+class ButtonCapsInfo:
+	buttonCaps: hidpi.HIDP_VALUE_CAPS
+	relativeIndexInCollection: int = 0
+
+
+class BrailleDisplayDriver(braille.BrailleDisplayDriver):
+	_dev: hwIo.hid.Hid
+	name = "hid"
+	# Translators: The name of a series of braille displays.
+	description = _("Standard HID Braille Display")
+	isThreadSafe = True
+
+	def __init__(self, port="auto"):
+		super(BrailleDisplayDriver, self).__init__()

```

Comment:
Can't you use Python 3 `super` here? The same goes for all `super` calls further down.